src/remoteClient/clientContextImpl.cpp

```diff
@@ -3957,7 +3957,7 @@ class InternalClientContextImpl :
     static size_t num_instances;
 
     InternalClientContextImpl(const Configuration::shared_pointer& conf) :
-        m_addressList(""), m_autoAddressList(true), m_connectionTimeout(30.0f), m_beaconPeriod(15.0f),
+        m_addressList(""), m_autoAddressList(true), m_connectionTimeout(30.0f), m_beaconPeriod(15.0f), m_maxBeacons(10), m_maxBeaconLifetime(15),
```

Comment:
A limit of 10 servers, and a 15 second lifetime seem quite small.  Have you checked one a network with more than 10 PVA servers to see how often `ChannelSearchManager::newServerDetected()` is called?